sources/packages/web/src/services/InstitutionUserService.ts

```diff
@@ -0,0 +1,250 @@
+        );
+    }
+
+    return permissions;
+  }
+
+  /**
+   * Create a user, associate with the institution, and assign the authorizations.
+   * @param bceidUserId user BCeID id from BCeID Web Service (e.g. SomeUserName)
+   * that will have its data retrieved to be created on SIMS.
+   * @param isAdmin must be created as an admin user.
+   * @param isLegalSigningAuthority for admin users, define if the role
+   * legal-signing-authority should be associated with the user.
+   * @param locationAuthorizations for non-admin users, the individual permission
+   * for every location.
+   * @param institutionId institution to have the user associated. If not provided the
+   * token information will be used, if available.
+   */
+  async createInstitutionUserWithAuth(
+    bceidUserId: string,
+    isAdmin: boolean,
+    isLegalSigningAuthority: boolean,
+    locationAuthorizations: LocationAuthorization[],
+    institutionId?: number,
+  ): Promise<void> {
+    const userPayload = { bceidUserId } as CreateInstitutionUserAPIInDTO;
+    userPayload.permissions = this.createUserPermissions(
+      isAdmin,
+      isLegalSigningAuthority,
+      locationAuthorizations,
+    );
+    await ApiClient.InstitutionUserApi.createInstitutionUserWithAuth(
+      userPayload,
+      institutionId,
+    );
+  }
+
+  /**
+   * Updates an existing user authorizations.
+   * @param institutionUserId institution user id to have the permissions updated.
+   * @param isAdmin must be created as an admin user.
+   * @param isLegalSigningAuthority for admin users, define if the role
+   * legal-signing-authority should be associated with the user.
+   * @param locationAuthorizations for non-admin users, the individual permission
+   * for every location.
+   */
+  async updateInstitutionUserWithAuth(
+    institutionUserId: number,
+    isAdmin: boolean,
+    isLegalSigningAuthority: boolean,
+    locationAuthorizations: LocationAuthorization[],
+  ): Promise<void> {
+    const userPayload = {} as UpdateInstitutionUserAPIInDTO;
+    userPayload.permissions = this.createUserPermissions(
+      isAdmin,
+      isLegalSigningAuthority,
+      locationAuthorizations,
+    );
+    await ApiClient.InstitutionUserApi.updateInstitutionUserWithAuth(
+      institutionUserId,
+      userPayload,
+    );
+  }
+
+  /**
+   * Get institution user by institution user id.
+   * @param institutionUserId institution user id to have the permissions updated.
+   * @returns institution user details.
+   */
+  async getInstitutionUserById(
+    institutionUserId: number,
+  ): Promise<InstitutionUserAPIOutDTO> {
+    return ApiClient.InstitutionUserApi.getInstitutionUserById(
+      institutionUserId,
+    );
+  }
+
+  /**
+   * Update the active status of the user.
+   * @param institutionUserId institution user id to have the permissions updated.
+   * @param isActive enable or disable the user.
+   */
+  async updateUserStatus(institutionUserId: number, isActive: boolean) {
```

Comment:
not part of this PR, but its missing `: Promise<void> {`

Reply:
I will add it.

Reply:
Added!